artemis-protocols/artemis-stomp-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/stomp/StompSession.java

```diff
@@ -154,12 +154,16 @@ public int sendMessage(MessageReference ref,
             encoder.encode(buffer, bodySize);
             encoder.close();
          } else {
-            buffer = coreMessage.getReadOnlyBodyBuffer();
+            if (Boolean.TRUE.equals(serverMessage.getBooleanProperty(Message.HDR_LARGE_COMPRESSED))) {
+               buffer = coreMessage.getBodyBuffer();
```

Comment:
@clebertsuconic Here I use body buffer instead of readonly bodybuffer, as I need to decompress the buffer and write into the message before delivering (like in large message case). Don't know how to do it a better way?

Reply:
@gaohoward I will send a PR... the same code you wrote would work with readOnlyBuffer...

Besides this problem is not just stomp.. AMQP and MQTT would fail. I will make this generic.

Reply:
@clebertsuconic Thanks!